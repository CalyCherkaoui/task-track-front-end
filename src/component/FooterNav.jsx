import React from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { IconContext } from 'react-icons';
import { AiOutlineAreaChart } from 'react-icons/ai';
import { CgExtensionAdd } from 'react-icons/cg';
import { GoDiffAdded } from 'react-icons/go';
import { ImLab } from 'react-icons/im';
import { Navbar, Nav } from 'react-bootstrap';
import styles from '../styles/Navigation.module.css';

const FooterNav = () => {
  const { isLoggedIn } = useSelector((state) => state.authentication);
  const { admin } = useSelector((state) => state.authentication);

  return (

    <Navbar className="justify-content-between">
      <Nav className={styles.footer_nav_wrapper}>
        {
            (isLoggedIn) ? (
              <div className="navbar-nav ml-auto">
                <li className="nav-item">
                  <Link to="/addtask" className="nav-link">
                    <span className={styles.nav_logo_emph}>
                      <IconContext.Provider value={{ className: 'footer_icon' }}>
                        <AiOutlineAreaChart />
                      </IconContext.Provider>
                    </span>
                    Track Task
                  </Link>
                </li>
                <li className="nav-item">
                  <Link to="/timeline" className="nav-link">
                    <span className={styles.nav_logo_emph}>
                      <IconContext.Provider value={{ className: 'footer_icon' }}>
                        <GoDiffAdded />
                      </IconContext.Provider>
                    </span>
                    Timeline
                  </Link>
                </li>
                <li className="nav-item">
                  <Link to="/measureup" className="nav-link">
                    <span className={styles.nav_logo_emph}>
                      <IconContext.Provider value={{ className: 'footer_icon' }}>
                        <ImLab />
                      </IconContext.Provider>
                    </span>
                    Measure-up
                  </Link>
                </li>
              </div>
            ) : (
              <div />
            )
          }
        {
            (admin && isLoggedIn) ? (
              <li className="nav-item">
                <Link to="/routineform" className="nav-link">
                  <span className={styles.nav_logo_emph}>
                    <IconContext.Provider value={{ className: 'footer_icon' }}>
                      <CgExtensionAdd />
                    </IconContext.Provider>
                  </span>
                  add routine
                </Link>
              </li>
            )
              : <div />
          }
      </Nav>
    </Navbar>

  );
};

export default FooterNav;
