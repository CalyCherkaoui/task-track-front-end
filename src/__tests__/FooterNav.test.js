import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store';
import FooterNav from '../component/FooterNav';

describe(
  'Renders correctly The Footer navigation component',
  () => {
    it(
      'Matches Footer Navigation snapshot',
      () => {
        const rendered = renderer.create(
          <Provider store={store}>
            <Router>
              <FooterNav />
            </Router>
          </Provider>,
        );

        expect(rendered).toMatchSnapshot();
      },
    );
  },
);
