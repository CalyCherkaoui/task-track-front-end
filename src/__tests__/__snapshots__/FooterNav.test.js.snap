// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly The Footer navigation component Matches Footer Navigation snapshot 1`] = `
<div
  className="footer_wrapper box_flex_col_centered"
>
  <div
    className="g-0 container"
  />
</div>
`;
